src/sinks/azure_blob/config.rs

```diff
@@ -50,6 +50,17 @@ pub struct AzureBlobSinkConfig {
     /// [az_cli_docs]: https://docs.microsoft.com/en-us/cli/azure/account?view=azure-cli-latest#az-account-get-access-token
     pub storage_account: Option<String>,
 
+    /// The Azure Blob Storage Endpoint URL.
+    ///
+    /// This is used to override the default blob storage endpoint URL in cases where you are using
+    /// credentials read from the environment/managed identities or access tokens without using an
+    /// explicit connection_string (which already explicitly supports overriding the blob endpoint
+    /// URL).
+    ///
+    /// This may only be used with `storage_account` and will be ignored when used with
+    /// `connection_string`.
+    pub blob_endpoint: Option<String>,
```

Comment:
The common convention for a config parameter like this would be simply `endpoint`. I think the other `blob_` prefixed names are specific to the individual blob stored (ie filename modifiers).

Reply:
I'm okay with renaming it to `endpoint` but it feels somewhat weird because Azure has many endpoints. This one is specific for blob storage and matches the name you'd find in Azure documentation. It is also the name of the k/v that is passed in to the `connection_string` when using a SAS/access key.

https://learn.microsoft.com/en-us/connectors/azureblob/#get-the-azure-storage-blob-endpoint-for-a-given-storage-account

Reply:
I'll defer to @bruceg here, but the context and namespace here is Azure blob storage, so it does feel redundant.

Reply:
That was my thinking as well. Any other Azure endpoints would be served by a different _type_ of sink, and we are already in the context of the "blob" service. In the same way, we have many different types of AWS and GCP sinks, but they all have just an optional `endpoint` option (albeit there with the complication of a `region` setting for AWS, but that's a separate matter) rather than a `s3_endpoint`, `gcs_endpoint`, etc.

Reply:
Alright! I will make the change to name. Will push that either a little later today or tomorrow morning.

Thanks!

Reply:
Things got away from me last week, but it should all be resolved now!